src/ert/shared/plugins/hook_specifications/forward_model_steps.py

```diff
@@ -0,0 +1,42 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Dict, List, Optional, Type, no_type_check
+
+from ert.config import ForwardModelStepPlugin
+from ert.shared.plugins.plugin_manager import hook_specification
+
+if TYPE_CHECKING:
+    from ert.shared.plugins.plugin_response import PluginResponse
+
+
+@no_type_check
+@hook_specification
+def installable_forward_model_steps() -> (
+    PluginResponse[List[Type[ForwardModelStepPlugin]]]
+):
+    """
+    :return: List of forward model step plugins in the form of subclasses of the
+        ForwardModelStepPlugin class
+    :rtype: PluginResponse with data as List[Type[ForwardModelStepPlugin]]
+    """
+
+
+@no_type_check
+@hook_specification(firstresult=True)
+def forward_model_step_documentation(
+    forward_model_step_name: str, description: str, examples: str, category: str
+) -> PluginResponse[Optional[Dict[str, str]]]:
+    """
+    :return: If forward_model_step_name is from your plugin return
+             dict with documentation fields as keys and corresponding
+             text as value (See below for details), else None.
+
+    Valid fields:
```

Comment:
I don't think this is valid google style docstring: https://www.sphinx-doc.org/en/master/usage/extensions/example_google.html#example-google so it won't be possible to put it into the docs. Which is something we should do in order to document how to make a plugin for third parties.

Reply:
Fixed, the comment now corresponds to the google docstring style

Reply:
We will create an issue to get the plugin docstrings into the docs after this then.